xdcs-server/src/main/java/pl/edu/agh/xdcs/services/sweeper/TaskSweeper.java

```diff
@@ -116,16 +128,39 @@ private void startTask(QueuedTaskEntity task, RuntimeTaskEntity runtimeTask, Lis
                                     .setAgentCount(agents.size())
                                     .setAgentId(agentId)
                                     .build()
-                    )
-                    .build());
+                    );
+            if (shouldContainKernelConfig(task.getHistoricalTask())) {
+                taskSubmit.setKernelConfig(kernelConfig);
+            }
+            taskRunner.submit(taskSubmit.build());
             agentId++;
         }
     }
 
+    private KernelConfig prepareKernelConfig(HistoricalTaskEntity task) {
+        KernelConfig.Builder kernelConfig = KernelConfig.newBuilder()
+                .addAllGlobalWorkShape(WorkShape.asList(task.getGlobalWorkShape()))
+                .addAllKernelArguments(
+                        new TreeMap<>(task.getKernelArguments())
+                                .values().stream()
+                                .map(ObjectRefEntity::getReferencedObjectId)
+                                .collect(Collectors.toList()));
+        if (task.getLocalWorkShape() != null) {
+            kernelConfig.addAllLocalWorkShape(WorkShape.asList(task.getLocalWorkShape()));
+        }
+        return kernelConfig.build();
+    }
+
     private Collection<String> getAgentIps(Set<AgentEntity> agents) {
         return agents.stream()
                 .map(AgentEntity::getAddress)
                 .map(InetAddress::getHostAddress)
                 .collect(Collectors.toList());
     }
+
+    private boolean shouldContainKernelConfig(HistoricalTaskEntity task) {
```

Comment:
to można dodać jako metoda do interfejsu `Task` i nazwać ją `isKernelExecutionTask`

Reply:
zrobione